NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    	    end
+
+    	    ScenarioFramework.AttachUnitsToTransports(attack_units:GetPlatoonUnits(), {transport})
+    	    WaitSeconds(0.5)
+
+    		IssueTransportUnload({transport}, ScenarioUtils.MarkerToPosition('M2_Transport_Drop_' .. i))
+
+    		IssueMove({transport}, ScenarioUtils.MarkerToPosition('UEF_Main_Base'))
+
+    		attack_units.PlatoonData = {}
+    		attack_units.PlatoonData.PatrolChain = ('M2_Transport_Attack_Route')
+    		ScenarioPlatoonAI.PatrolThread(attack_units)
+    	end
+
+    	for i = 1, 2 do
+    	   	local transport = ScenarioUtils.CreateArmyUnit('UEF', 'M2_Transport')
+
+    	    for _, v in base_units do
+    	        table.insert(allUnits, v)
+    	    end
+
+    	    ScenarioFramework.AttachUnitsToTransports(base_units:GetPlatoonUnits(), {transport})
+    	    WaitSeconds(0.5)
+
+    		IssueTransportUnload({transport}, ScenarioUtils.MarkerToPosition('M2_Transport_Base_Drop_' .. i))
+
+    		IssueMove({transport}, ScenarioUtils.MarkerToPosition('UEF_Main_Base'))
+
+    		base_units.PlatoonData = {}
+    		base_units.PlatoonData.PatrolChain = ('M2_Outpost_Patrol')
+    		ScenarioPlatoonAI.PatrolThread(base_units)
+    	end
+    end
+end
+
+function M2()
+	----------
+	-- Objectives
+	----------
+	ScenarioInfo.M2P1 = Objectives.CategoriesInArea(
+    'primary',                      -- type
+    'incomplete',                   -- complete
+    'Destroy Enemy Outpost',  -- title
+    'Enemy forces are hammering our position! Destroy the enemy outpost to soften the attacks.',  -- description
+    'kill',
+    {
+        MarkUnits = false,
+        Requirements = {
+            { Area = 'M1Outpost', Category = categories.FACTORY + categories.ENGINEER, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+        },
+    })
+    table.insert(AssignedObjectives, ScenarioInfo.M2P1)
+
+    ScenarioInfo.M2P2 = Objectives.KillOrCapture(
+    'primary',                      -- type
+    'incomplete',                   -- complete
+    'Destroy Tactical Missile Launcher',  -- title
+    'The Enemy Commander has a Missile Launcher firing at our ship. Destroy it before we take too much damage.',  -- description
+        {                               -- target
+            Units = {ScenarioInfo.TacLauncher}
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M2P2)
+
+    ScenarioInfo.M2P1:AddResultCallback(
+    function(result)
+        if(result) then
+            ScenarioFramework.Dialogue(OpStrings.UEFOutpost_Dead, nil, true)
+            Cinematics.EnterNISMode()
+            Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_02b'), 1)
+            WaitSeconds(4)
+            Cinematics.ExitNISMode()
+        	ForkThread(M2Capture)
+        end
+    end)
+end
+
+function M2Capture()
+    ----------
+    -- Spawn Base
+    ----------
+    local M3Walls = ScenarioUtils.CreateArmyGroup('UEF', 'M3Walls')
+    local Engineers = ScenarioUtils.CreateArmyGroup('UEF', 'M3Engineers')

```

Comment:
are these the base engineers? if so they could just be part of the base, spawned with all the buildings